data-access/src/object_store/mod.rs

```diff
@@ -91,6 +92,39 @@ pub trait ObjectStore: Sync + Send + Debug {
         })))
     }
 
+    /// Calls `list_file` with a glob_pattern
+    async fn glob_file(&self, path: &str) -> Result<FileMetaStream> {
+        const GLOB_CHARS: [char; 7] = ['{', '}', '[', ']', '*', '?', '\\'];
+
+        /// Determine whether the path contains a globbing character
+        fn is_glob_path(path: &str) -> bool {
+            path.chars().any(|c| GLOB_CHARS.contains(&c))
+        }
+
+        if !is_glob_path(path) {
+            self.list_file(path).await
+        } else {
+            // take path up to first occurence of a glob char
+            let path_to_first_glob_character: Vec<&str> =
+                path.splitn(2, |c| GLOB_CHARS.contains(&c)).collect();
+            // find last occurrence of folder /
+            let path_parts: Vec<&str> = path_to_first_glob_character[0]
```

Comment:
I'm not intimately familiar with the Rust filesystem APIs, but relying on a '/' path separator feels like it might cause excitement on windows.

Perhaps this could use https://doc.rust-lang.org/std/path/struct.Path.html#method.parent instead?